src/ert/gui/model/snapshot.py

```diff
@@ -115,20 +115,23 @@ def prerender(
             REAL_STATUS_COLOR: defaultdict(dict),
         }
 
+        for real_id, real in reals.items():
+            if real.status:
+                metadata[REAL_STATUS_COLOR][real_id] = _QCOLORS[
+                    state.REAL_STATE_TO_COLOR[real.status]
+                ]
+
         if isinstance(snapshot, Snapshot):
             metadata[SORTED_REALIZATION_IDS] = sorted(snapshot.reals.keys(), key=int)
             metadata[SORTED_JOB_IDS] = defaultdict(list)
-            for (real_id, job_id), _ in job_states.items():
+            for (real_id, job_id), job_status in job_states.items():
                 metadata[SORTED_JOB_IDS][real_id].append(job_id)
-
-        for (real_id, job_id), job_status in job_states.items():
-            if real_id in reals and reals[real_id].status:
-                metadata[REAL_STATUS_COLOR][real_id] = _QCOLORS[
-                    state.REAL_STATE_TO_COLOR[reals[real_id].status]
-                ]
-
-            color = _QCOLORS[state.JOB_STATE_TO_COLOR[job_status]]
-            metadata[REAL_JOB_STATUS_AGGREGATED][real_id][job_id] = color
+                color = _QCOLORS[state.JOB_STATE_TO_COLOR[job_status]]
+                metadata[REAL_JOB_STATUS_AGGREGATED][real_id][job_id] = color
+        else:
+            for (real_id, job_id), job_status in job_states.items():
+                color = _QCOLORS[state.JOB_STATE_TO_COLOR[job_status]]
+                metadata[REAL_JOB_STATUS_AGGREGATED][real_id][job_id] = color
```

Comment:
Personally i'm not overjoyed with duplicate code loops, but the immediate alternative isn't really that much better.
Maybe it might be more readable but a teeny tiny less efficient due to the `is_snapshot` bool check.

```diff
+        is_snapshot = False
+
         if isinstance(snapshot, Snapshot):
             metadata[SORTED_REALIZATION_IDS] = sorted(snapshot.reals.keys(), key=int)
             metadata[SORTED_JOB_IDS] = defaultdict(list)
-            for (real_id, job_id), job_status in job_states.items():
+            is_snapshot = True
+
+        for (real_id, job_id), job_status in job_states.items():
+            metadata[SORTED_JOB_IDS][real_id].append(job_id) if is_snapshot else None
-                color = _QCOLORS[state.JOB_STATE_TO_COLOR[job_status]]
-                metadata[REAL_JOB_STATUS_AGGREGATED][real_id][job_id] = color
-        else:
-            for (real_id, job_id), job_status in job_states.items():
-                color = _QCOLORS[state.JOB_STATE_TO_COLOR[job_status]]
-                metadata[REAL_JOB_STATUS_AGGREGATED][real_id][job_id] = color
+            color = _QCOLORS[state.JOB_STATE_TO_COLOR[job_status]]
+            metadata[REAL_JOB_STATUS_AGGREGATED][real_id][job_id] = color
```

Reply:
My reasoning was that there is a lot more partial snapshots than full snapshots and therefore one could avoid the extra if-checks in the for-loop. However, I do agree that it is more confusing to read and probably not that much to gain from it, so I will edit this.